src/prefect/engine/task_runner.py

```diff
@@ -613,6 +616,39 @@ def load_results(
         """
         return state, upstream_states
 
+    def check_target(self, state: State, inputs: Dict[str, Result]) -> State:
+        """
+        Checks if a Result exists at the task's target.
+
+        Args:
+            - state (State): the current state of this task
+            - inputs (Dict[str, Result]): a dictionary of inputs whose keys correspond
+                to the task's `run()` arguments.
+
+        Returns:
+            - State: the state of the task after running the check
+        """
+        result = self.task.result
+        target = self.task.target
+
+        if result and target:
+            if result.exists(target):
+                # expiration = pendulum.now("utc") + self.task.cache_for
+                # TODO: How to handle cached durations?
```

Comment:
Should we introduce the ability to allow for infinite horizon caches?

Reply:
Yeah we should for this because it doesn't necessarily have a _duration_ but the duration is until the target does not exist

Reply:
@joshmeek Whenever you create the `Cached` state, if you set `cached_result_expiration=None` then the `duration_only` validator will always return `True`, meaning the cache will effectively last for forever (until your pipeline step says otherwise at least).